src/hdmf/common/table.py

```diff
@@ -470,37 +488,47 @@ def __eq__(self, other):
             {'name': 'index', 'type': (bool, VectorIndex, 'array_data'),
              'doc': 'whether or not this column should be indexed', 'default': False})
     def add_column(self, **kwargs):
-        name = getargs('name', kwargs)
-        for col in self.__columns__:
-            if col['name'] == name:  # column has not been added but is pre-specified
-                msg = "column '%s' already exists in %s '%s'" % (name, self.__class__.__name__, self.name)
-                raise ValueError(msg)
-
-        self._add_column(**kwargs)
-
-    @docval({'name': 'name', 'type': str, 'doc': 'the name of this VectorData'},
-            {'name': 'description', 'type': str, 'doc': 'a description for this column'},
-            {'name': 'data', 'type': ('array_data', 'data'),
-             'doc': 'a dataset where the first dimension is a concatenation of multiple vectors', 'default': list()},
-            {'name': 'table', 'type': (bool, 'DynamicTable'),
-             'doc': 'whether or not this is a table region or the table the region applies to', 'default': False},
-            {'name': 'index', 'type': (bool, VectorIndex, 'array_data'),
-             'doc': 'whether or not this column should be indexed', 'default': False})
-    def _add_column(self, **kwargs):
         """
         Add a column to this table.
 
         If data is provided, it must contain the same number of rows as the current state of the table.
 
-        :raises ValueError
+        :raises ValueError: if the column has already been added to the table
         """
-        name, data = getargs('name', 'data', kwargs)
+        name, data, description = getargs('name', 'data', 'description', kwargs)
         index, table = popargs('index', 'table', kwargs)
 
+        if isinstance(index, VectorIndex):
```

Comment:
@ajtritt In what situation would a user pass a `VectorIndex` to `add_column`? When this happens, `add_column` creates a new `VectorData` column for the given name, description, and data, and also adds the `VectorIndex` which must have a different target `VectorData`, because its target is required in the constructor. I added this warning because I could not figure out how one would use this. It looks intentional too since there is code below that checks `if isinstance(index, VectorIndex):` and `if not isinstance(col_index.parent, Container):`

Reply:
I am not sure, but I think the warning is appropriate.